Sources/Public/Item Views/SwiftUIWrapperView.swift

```diff
@@ -0,0 +1,198 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+// MARK: - SwiftUIWrapperView
+
+/// A wrapper view that enables SwiftUI `View`s to be used with `CalendarItemModel`s.
+@available(iOS 13.0, *)
+public final class SwiftUIWrapperView<Content: View>: UIView {
+
+  // MARK: Lifecycle
+
+  public init(content: Content) {
+    self.content = content
+
+    super.init(frame: .zero)
+
+    insetsLayoutMarginsFromSafeArea = false
+    layoutMargins = .zero
+  }
+
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: Public
+
+  public override func didMoveToWindow() {
+    super.didMoveToWindow()
+
+    if window != nil {
+      setUpHostingControllerIfNeeded()
+    }
+  }
+
+  public override func layoutSubviews() {
+    super.layoutSubviews()
+    hostingControllerView?.frame = bounds
+  }
+
+  // MARK: Fileprivate
+
+  fileprivate var content: Content {
+    didSet {
+      hostingController.rootView = content
+      configureGestureRecognizers()
+    }
+  }
+
+  // MARK: Private
+
+  private lazy var hostingController = HostingController<Content>(rootView: content)
+
+  private weak var hostingControllerView: UIView?
+
+  private func setUpHostingControllerIfNeeded() {
+    guard let closestViewController = closestViewController() else {
+      assertionFailure(
+        "Could not find a view controller to which the `UIHostingController` could be added.")
+      return
+    }
+
+    guard hostingController.parent !== closestViewController else { return }
+
+    if hostingController.parent != nil {
+      hostingController.willMove(toParent: nil)
+      hostingController.view.removeFromSuperview()
+      hostingController.removeFromParent()
+      hostingController.didMove(toParent: nil)
+    }
+
+    hostingController.willMove(toParent: closestViewController)
+    closestViewController.addChild(hostingController)
+    hostingControllerView = hostingController.view
+    addSubview(hostingController.view)
+    hostingController.didMove(toParent: closestViewController)
+
+    setNeedsLayout()
+  }
+
+  // This allows touches to be passed to `ItemView` even if the SwiftUI `View` has a gesture
+  // recognizer.
+  private func configureGestureRecognizers() {
+    for gestureRecognizer in hostingControllerView?.gestureRecognizers ?? [] {
+      gestureRecognizer.cancelsTouchesInView = false
+    }
+  }
+
+}
+
+// MARK: CalendarItemViewRepresentable
+
+@available(iOS 13.0, *)
+extension SwiftUIWrapperView: CalendarItemViewRepresentable {
+
+  public struct InvariantViewProperties: Hashable {
+
+    // MARK: Lifecycle
+
+    init(initialContent: Content) {
+      self.initialContent = initialContent
+    }
+
+    // MARK: Public
+
+    public static func == (lhs: InvariantViewProperties, rhs: InvariantViewProperties) -> Bool {
+      type(of: lhs.initialContent) == type(of: rhs.initialContent)
+    }
```

Comment:
Since the `InvariantViewProperties` is nested within the `SwiftUIWrapperView` type which is genercized on `Content` won't this always be true?

Reply:
Great point, yes!